angular_forms/lib/src/directives/validators.dart

```diff
@@ -172,11 +172,10 @@ class MaxLengthValidator implements Validator {
   ],
 )
 class PatternValidator implements Validator {
-  final ValidatorFn _validator;
-
-  PatternValidator(@Attribute('pattern') String pattern)
-      : _validator = Validators.pattern(pattern);
+  @HostBinding('attr.pattern')
+  @Input()
+  String pattern;
 
   @override
-  Map<String, dynamic> validate(AbstractControl c) => _validator(c);
+  Map<String, dynamic> validate(AbstractControl c) => Validators.pattern(pattern)(c);
```

Comment:
Probably too much of a change for this PR, but it would be nice if we didn't have to re-parse the regex on every validate call.

Reply:
Yes, it would be too much for this PR